File: app/src/main/java/shts/jp/android/nogifeed/utils/PicassoHelper.java
package shts.jp.android.nogifeed.utils;

import android.content.Context;
import android.widget.ImageView;
import android.widget.RemoteViews;

import com.squareup.picasso.Picasso;

import shts.jp.android.nogifeed.views.transformations.CircleTransformation;

public class PicassoHelper {

    private static final CircleTransformation CIRCLE_TRANSFORMATION = new CircleTransformation();

    public static void load(Context context, ImageView target, String url) {
        Picasso.with(context)
                .load(url)
                .fit()
                .centerCrop()
                .into(target);
    }

    public static void loadAndCircleTransform(Context context, ImageView target, String url) {
        Picasso.with(context)
                .load(url)
                .fit()
                .centerCrop()
                .transform(CIRCLE_TRANSFORMATION)
                .into(target);
    }

    public static void loadAndCircleTransform(Context context, String url, RemoteViews remoteViews,
                                              int resId, int appWidgetId) {
        Picasso.with(context)
                .load(url)
                .transform(CIRCLE_TRANSFORMATION)
                .into(remoteViews, resId, new int[] { appWidgetId });
    }

}

File: app/src/main/java/shts/jp/android/nogifeed/utils/PicassoHelper.kt
package shts.jp.android.nogifeed.utils

import android.content.Context
import android.support.annotation.DrawableRes
import android.text.TextUtils
import android.widget.ImageView

import com.squareup.picasso.Picasso

import shts.jp.android.nogifeed.views.transformations.CircleTransformation

object PicassoHelper {

    private val CIRCLE_TRANSFORMATION = CircleTransformation()

    fun load(context: Context, target: ImageView, url: String) {
        Picasso.with(context)
                .load(url)
                .fit()
                .centerCrop()
                .into(target)
    }

    fun load(target: ImageView, url: String?, @DrawableRes fallback: Int) {
        if (TextUtils.isEmpty(url)) {
            target.setImageResource(fallback)
        } else {
            Picasso.with(target.context)
                    .load(url)
                    .fit()
                    .centerCrop()
                    .into(target)
        }
    }

    fun loadAndCircleTransform(context: Context, target: ImageView, url: String) {
        Picasso.with(context)
                .load(url)
                .fit()
                .centerCrop()
                .transform(CIRCLE_TRANSFORMATION)
                .into(target)
    }

    fun loadAndCircleTransform(target: ImageView, url: String?, @DrawableRes fallback: Int) {
        if (TextUtils.isEmpty(url)) {
            target.setImageResource(fallback)
        } else {
            Picasso.with(target.context)
                    .load(url)
                    .fit()
                    .centerCrop()
                    .transform(CIRCLE_TRANSFORMATION)
                    .into(target)
        }
    }
}
